### PR TITLE
chore: use new FAB functionality for component schemas update OAS

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY

FAB 3.4.3 now implements `openapi_spec_component_schemas` ;) this PR removes that functionality from Superset.

Also the new version properly uses OpenAPI spec servers with actual servers and we can now define a list of servers for docs, by default the included swagger UI will set the current request host URL.

Since the spec is not using a base path (`api/v1`) anymore and FAB is not relying on the `resource_name` to infer the spec path but the `route_base` (this used to be a limitation). All endpoints are declared with their full relative path:

This PR also fixes the OpenAPI spec auto generation for Superset docs, so that all the schema components are not declared with `MetaN` anymore.
 
So these have are not duplicated anymore:

Screenshot 2022-01-07 at 16 12 40

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:
Screenshot 2022-01-07 at 16 11 03

After:
Screenshot 2022-01-07 at 16 09 13

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/base_api.py [hunk 1]
```diff
@@ -217,33 +213,19 @@ class BaseSupersetModelRestApi(ModelRestApi):
...
         # Add base API spec base query parameter schemas
         if self.apispec_parameter_schemas is None:  # type: ignore
             self.apispec_parameter_schemas = {}
         self.apispec_parameter_schemas["get_related_schema"] = get_related_schema
-        if self.openapi_spec_component_schemas is None:
-            self.openapi_spec_component_schemas = ()
-        self.openapi_spec_component_schemas = self.openapi_spec_component_schemas + (
+        self.openapi_spec_component_schemas: Tuple[
+            Type[Schema], ...
+        ] = self.openapi_spec_component_schemas + (
```
Comment (villebro) [inline superset/views/base_api.py:225]: Do we really need this type declaration? Seems like this should be enough:
```suggestion
        self.openapi_spec_component_schemas = self.openapi_spec_component_schemas + (
```

**Review by villebro — APPROVED** [review]
I much prefer this new UI! Also, great to see this logic being moved upstream! 🙂 LGTM

File: superset/views/base_api.py (hunk 1 above)
Comment (dpgaspar) [reply superset/views/base_api.py:225]: it does, and it surprised me also, this is declared on FAB here: https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/api/__init__.py#L297

Comment (stevetracvc) [reply superset/views/base_api.py:225]: I'm trying to merge master and getting an error on this line.

AttributeError: 'AnnotationRestApi' object has no attribute 'openapi_spec_component_schemas'

Why were lines 226 and 227 removed?
```
        if self.openapi_spec_component_schemas is None:
            self.openapi_spec_component_schemas = ()
```

**Pull request merged**